lib/assert.js

```diff
@@ -233,6 +233,9 @@ function parseCode(code, offset) {
   }
   let node;
   let start = 0;
+  const regExp = code.match(/^.*try.*?{/);
+  if (regExp != null)
+    start = regExp.index + regExp[0].length;
```

Comment:
There are multiple cases that won't fit into this pattern. Think about e.g., the assertion being before the `try / catch`.

There are also probably more things that might cause failures, not just `try / catch`. I personally feel like this needs a deeper refactoring.

There is indeed one simple fix that will always work but that would cause CPU spikes in some cases, so I do not even propose it ;-).

Reply:
I think there must be `try` keywords on the first line after reading the whole codes. Don't it?

Reply:
Yes, but it could be behind the assertion. Think about e.g., minified code. I can not think of any regular expression that guarantees that things work as expected. We need an AST for it.